CONTRIBUTING.md

```diff
@@ -0,0 +1,289 @@
+a discussion. Use the [`project-reactor`](https://stackoverflow.com/questions/tagged/project-reactor) tag for that purpose.
+
+If you prefer real-time discussion, we also have a [Gitter channel](https://gitter.im/reactor/reactor).
+
+## :beetle: Did you find a bug?
+
+ - **Do not open a GitHub issue if the bug is a security vulnerability in Reactor** (see [SECURITY.md](https://github.com/reactor/.github/SECURITY.md))
+ - **Ensure the bug was not already reported**: Do a bit of searching
+the relevant project's **Issues** (eg. [reactor-core Issues](https://github.com/reactor/reactor-core/issues/)) to see if you
+can find something similar
+ - If you don't find something similar, **open a new issue** (eg. [reactor-core new issue](https://github.com/reactor/reactor-core/issues/new)).
+ Be sure to follow the template and to include a title and clear description.
+ Add as much relevant information as possible, and a code sample or an executable test case demonstrating the expected behavior that is not occurring
+
+## :wrench: Did you write a patch that fixes a bug?
+
+ - _Unless the fix is trivial_ (typo, cosmetic changes that don't modify the behavior, ...), there **should be an issue** associated with it.See [Did you find a bug?](#beetle-did-you-find-a-bug).
+ - Wait for **issue triage** before you start coding. This will determine from which branch to start. See [Starting from the right branch](#starting-from-the-right-branch) section for more details
+ - Be sure to **[sign](https://cla.pivotal.io/sign/spring) the Contributor Licence Agreement(CLA)**. A bot will remind you about that anyway. The CLA form integrates with GitHub, just follow the steps
+ - Try to create a branch with a **meaningful name** (see hints [here](#creating-a-branch-with-representative-name))
+ - Work on your change. Be sure to include **JUnit test cases**, this will greatly help maintainers while reviewing your change
+ - **Run all tests locally** prior to submission: `./gradlew check`
+ - Finally, you're **ready to submit** your Pull-Request :+1:
+
+## :mag: Do you intend to add a new feature or change an existing one?
+ - Again, **search for similar suggestions** in existing issues (including closed ones)
+ - If nothing like that has been suggested before, **discuss it in a new issue first**
+ - If you gather positive feedback, **work on the change** following the guideline [above](#wrench-did-you-write-a-patch-that-fixes-a-bug)
+
+## :art: Code style
+The Reactor code style can be [imported into your IDE](#importing-and-following-the-reactor-code-style).
+
+Please carefully follow the whitespace and formatting conventions already
+present in the codebase. Here is a short summary of the style:
+
+1. Tabs, not spaces
+1. Unix (LF), not DOS (CRLF) line endings
+1. Eliminate all trailing whitespace
+1. Wrap Javadoc at 90 characters
+1. Aim to wrap code at 90 characters, but favor readability over wrapping
```

Comment:
Since Spring is using 120 now, we should probably use 120 too

Reply:
fixed in 11e060d